services/server/src/handlers/availabilitySlot.ts

```diff
@@ -0,0 +1,205 @@
+import { bad, conflict, forbidden, notfound } from "@/lib/error";
+import { datetime, id, skippablePagination } from "@/validation/utils";
+import { isTutor, isTutorManager, isUser } from "@litespace/auth";
+import { IAvailabilitySlot } from "@litespace/types";
+import {
+  availabilitySlots,
+  interviews,
+  knex,
+  lessons,
+} from "@litespace/models";
+import dayjs from "@/lib/dayjs";
+import { NextFunction, Request, Response } from "express";
+import safeRequest from "express-async-handler";
+import { asSubSlots, isIntersecting } from "@litespace/sol";
+import zod from "zod";
+import { isEmpty } from "lodash";
+import { deleteSlots } from "@/lib/availabilitySlot";
+
+const findPayload = zod.object({
+  userId: id,
+  after: datetime,
+  before: datetime,
+  slotsOnly: zod.boolean().optional().default(false),
+  pagination: skippablePagination.optional(),
+});
+
+const setPayload = zod.object({
+  slots: zod.array(
+    zod.union([
+      zod.object({
+        action: zod.literal("create"),
+        start: datetime,
+        end: datetime,
+      }),
+      zod.object({
+        action: zod.literal("update"),
+        id: id,
+        start: datetime.optional(),
+        end: datetime.optional(),
+      }),
+      zod.object({
+        action: zod.literal("delete"),
+        id: id,
+      }),
+    ])
+  ),
+});
+
+async function find(req: Request, res: Response, next: NextFunction) {
+  const user = req.user;
+  if (!isUser(user)) return next(forbidden());
+
+  const { userId, after, before, slotsOnly, pagination } = findPayload.parse(
```

Comment:
As for the time being, it is not required to control the toggle of subslots on and off for outside. It might change in the future but not now. In this case we can remove this flag. 

```suggestion
  const { userId, after, before, pagination } = findPayload.parse(
```